tabu.sty

```diff
@@ -1298,6 +1299,58 @@
 \def\tabu@endofcollect #1{\ifnum0=`{}\fi
                           \expandafter\endgroup \the\toks@  #1%
 }% \tabu@endofcollect
+\catcode`\| \z@
+\catcode`\\ 12
+|long|def|tabu@collectbody@forscan #1#2#3\end{%
+    |def|tabu@collectbody@forscan@next{|tabu@collectbody@forscan{#1}}%
+    |def|tabu@collectbody@forscan@partial{#2#3}%
+    |futurelet|tabu@temp@token|tabu@collectbody@forscan@a
+}%
+|def|tabu@collectbody@forscan@a{%
+    |ifcat|tabu@temp@token e%
+        |expandafter|@firstoftwo
+    |else
+        |expandafter|@secondoftwo
+    |fi
+    |tabu@collectbody@forscan@aa
+    |tabu@collectbody@forscan@b
+}%
+|def|tabu@collectbody@forscan@aa #1{%
+    |expandafter|tabu@collectbody@forscan@next|expandafter{|tabu@collectbody@forscan@partial\end#1}%
+}%
+|def|tabu@collectbody@forscan@b #1{%
+    |edef|tabu@stack{|expandafter|tabu@pushbegins@forscan
+                     |tabu@collectbody@forscan@partial\begin|relax|expandafter|@gobble|tabu@stack}%
+    |ifx |tabu@stack|@empty
+        |expandafter|@firstoftwo
+    |else
+        |expandafter|@secondoftwo
+    |fi
+    {|expandafter|tabu@collectbody@forscan@end|expandafter{|tabu@collectbody@forscan@partial}{#1}}%
+    {|expandafter|tabu@collectbody@forscan@continue|expandafter{|tabu@collectbody@forscan@partial}{#1}}%
+}%
+|def|tabu@collectbody@forscan@end #1#2{%
+    |toks@|expandafter{|expandafter|tabu@thebody|expandafter{|the|toks@ #1}%
+            |def|tabu@end@envir{|end{#2}}%
+            |iftabu@long |def|tabu@endenvir {|end{#2}|tabu@gobbleX}%
+            |else        |def|tabu@endenvir {|let|endarray |@empty
+                                             |end{#2}|tabu@gobbleX}%
+            |fi}%
+    |let|tabu@collectbody@forscan|tabu@endofcollect
+    |tabu@collectbody@forscan@next
+}%
+|def|tabu@collectbody@forscan@continue #1#2{%
+    |def|tabu@temp{#2}%
+    |ifx |tabu@temp|@empty |toks@|expandafter{|the|toks@ #1\end }%
```

Comment:
this is causing me worries I don't quite see if I should leave or not a space after the `\end` (recall here ``\`` has catcode 12) because I don't quite understand what this does in parallel original `\tabu@collectbody` context

Reply:
I have removed the space here after `\end`, no idea why I had inserted it to start with. Sorry.